_posts/2016-08-09-collections-app.md

```diff
@@ -0,0 +1,82 @@
+separately --- a task that could quickly become tedious. What if you want to share all of the
+project's grains as a single unit?
+
+Now that we have released the
+[Collections app](https://apps.sandstorm.io/app/s3u2xgmqwznz2n3apf30sm3gw1d85y029enw5pymx734cnk5n78h),
+we have a satisfying answer to that question; to share many grains at once, you add them to a collection.
+
+<img src="/news/images/collections-1.png"
+ title="a collection with a chatroom, a kanban board, a spreadsheet, and two git repos">
+
+A collection is a list of grains. Any collaborator with whom you share a collection gets access to
+all of the grains in it. On the flip side, when you remove a grain from the collection, your
+collaborators lose access to that grain. Moreover, when you revoke a collaborator from the
+collection, that (now former-) collaborator loses access to all of the collection's grains.
+
+And since a collection is itself a grain, sharing one works just like sharing anything else on
+Sandstorm, through the "Share access" button.
+
+<img src="/news/images/collections-2.png" title='the "Share access" dialog'>
+
+### Powerbox makes it possible
+
+The fact that we have implemented collections *as an app* may come as a surprise. Indeed, the notion
+of a collection might seem fundamental enough to deserve being baked-in as a core feature of
+Sandstorm. However, Sandstorm bakes in an even more fundamental notion: the idea that grains can
+refer to and coordinate with one another. The embodiment of this idea is the [powerbox](/how-it-works#powerbox),
+an interface
+--- mediated and auditable by the user --- through which grains can exchange capabilities.
+
+When you click the "Add grain" button in a collection, the collection in fact initiates a powerbox
+request. Sandstorm then asks you, the user, to choose a grain with which to fulfill the request.
+
+<img src="/news/images/collections-3.png" title="Making a powerbox request">
+
+Once you select a grain, the collection receives a reference to that grain. It can then use that
+reference to retrieve metadata such as the grain's icon and the the name of the app that created the
+grain. Crucially, it can also offer the reference to your collaborators, so that your single "Add
+grain" action can result in all of your collaborators receiving access to the grain.
+
+Other apps can use the powerbox in the same way. You could write an alternative implementation of
+collections, either by starting from scratch or by forking [ours](
+https://github.com/dwrensha/sandstorm-collections-app ). And apps whose primary purpose has nothing
+to do with collections can also benefit from the ability to request and offer grain references. For
+example, a chat room could be enhanced by the ability to embed Sandstorm-aware links to other
+grains. Indeed, the latest release of
+[Rocket.Chat](https://apps.sandstorm.io/app/vfnwptfn02ty21w715snyyczw0nqxkv3jvawcah10c6z7hj1hnu0) does just that.
+
```

Comment:
I generally feel that way, but if you write a concise version of it here, I think it would make the post more compelling. As short as:

> To give you a sense of the code flow, here is how Collections works with Sandstorm to enable this functionality. First, it makes a powerbox request for a UiView capability. Once the user approves, the app receives a token; it calls claimRequest() on the token, then save() to get a long-lasting token. When it's time to use the UiView, for example to get grain metadata via getViewInfo(), it calls restore() to get a live reference. To open the grain for the user to use it call offer(), which opens the grain without opening a new browser tab. You can read the full details in the [Sandstorm documentation](http://example.com/).

But yeah, maybe what you're saying is you/I/someone should write up those 5 bullet points somewhere into the docs so that this blog post can link to them.